docs/app-dev-context/activity-heartbeats.md

```diff
@@ -0,0 +1,16 @@
+An Activity Heartbeat is a ping from the Worker that is executing the Activity to the Temporal Cluster.
+Each Heartbeat informs the Temporal Cluster that the Activity Execution is making progress and the Worker has not crashed.
+If the Cluster does not receive a Heartbeat within a [Heartbeat Timeout](/concepts/what-is-a-heartbeat-timeout) time period, the Activity will be considered failed and another [Activity Task Execution](/concepts/what-is-an-activity-task-execution) may be scheduled according to the Retry Policy.
+
+Heartbeats may not always be sent to the Cluster—they may be throttled by the Worker.
```

Comment:
Let's add a more explicit explanation of how they are throttled. It actually may be important that heartbeats that happen during the throttling period are not discarded and the last details are actually sent at the end.

This is how I tried to highlight it in the original thread:

> After sending a heartbeat we set a timer. We "turn on" throttling and stop sending heartbeats until the timer fires. We still observe and remember the last heartbeat that you sent during the throttling period. When the timer fires, we send the last heartbeat that came in during the throttling period and set the timer again.

Reply:
👍 Sounds good